test_scripts/RC/ButtonPress/005_check_rpc_paramters_for_button_press.lua

```diff
@@ -0,0 +1,141 @@
+
+    commonTestCases:DelayedExp(commonRC.timeout)
+end
+
+local function SendButtonPressNegative(button_press_params, self)
+    local cid = self.mobileSession:SendRPC("ButtonPress", button_press_params)
+
+    EXPECT_HMICALL("Buttons.ButtonPress")
+    :Times(0)
+
+    EXPECT_RESPONSE(cid, { success = false, resultCode = "INVALID_DATA" })
+
+    commonTestCases:DelayedExp(commonRC.timeout)
+end
+
+--[[ Positive Scenario - check all positive climate names params]]
+climate_params = reset_climate_params()
+runner.Title("Preconditions")
+runner.Step("Clean environment", commonRC.preconditions)
+runner.Step("Start SDL, HMI, connect Mobile, start Session", commonRC.start)
+runner.Step("RAI, PTU", commonRC.rai_ptu)
+for _, button_name_value in pairs( climate_button_names ) do
+    climate_params.buttonPressMode = "SHORT"
+    climate_params.buttonName = button_name_value
+    runner.Title("Test - ButtonPress with buttonName " .. button_name_value)
+    runner.Step("ButtonPress_CLIMATE_" .. button_name_value .. "_SHORT", SendButtonPressPositive, {climate_params})
+    climate_params.buttonPressMode = "LONG"
+    runner.Step("ButtonPress_CLIMATE_" .. button_name_value .. "_LONG", SendButtonPressPositive, {climate_params})
+end
+
+--[[ Positive Scenario - check all positive radio names params]]
+radio_params = reset_radio_params()
+for _, button_name_value in pairs( radio_button_names ) do
+    radio_params.buttonPressMode = "SHORT"
+    radio_params.buttonName = button_name_value
+    runner.Title("Test - ButtonPress with buttonName " .. button_name_value)
+    runner.Step("ButtonPress_RADIO_" .. button_name_value .. "_SHORT", SendButtonPressPositive, {radio_params})
+    radio_params.buttonPressMode = "LONG"
+    runner.Step("ButtonPress_RADIO_" .. button_name_value .. "_LONG", SendButtonPressPositive, {radio_params})
+end
+
+--[[ Negative Scenario - invalid value of buttonName in mobile request]]
+climate_params = reset_climate_params()
+radio_params = reset_radio_params()
+climate_params.buttonName = "invalid_name"
+radio_params.buttonName = "invalid_name"
+runner.Title("Test - negative, invalid value of buttonName in mobile request")
+runner.Step("ButtonPress_CLIMATE", SendButtonPressNegative, {climate_params})
+runner.Step("ButtonPress_RADIO", SendButtonPressNegative, {radio_params})
+climate_params = reset_climate_params()
+radio_params = reset_radio_params()
+
+--[[ Negative Scenario - invalid value of buttonPressMode in mobile request]]
+climate_params.buttonPressMode = "invalid_name"
+radio_params.buttonPressMode = "invalid_name"
+runner.Title("Test - negative, invalid value of buttonPressMode in mobile request")
+runner.Step("ButtonPress_CLIMATE", SendButtonPressNegative, {climate_params})
+runner.Step("ButtonPress_RADIO", SendButtonPressNegative, {radio_params})
+climate_params = reset_climate_params()
+radio_params = reset_radio_params()
+
+--[[ Negative Scenario - not matched params in mobile request]]
```

Comment:
accepted, corrected in https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/1574/commits/aabf070a85f944de0194ca580cc19b147cde8e17